posydon/interpolation/profile_interpolation.py

```diff
@@ -0,0 +1,553 @@
+        valid_initial = np.log10(np.array(valid_linear_initial))
+        valid_finalmass = self.valid_scalars["final_mass"].astype(np.float64)
+        
+        # instantiate and train H mass fraction profile model
+        h_ind = self.names.index("x_mass_fraction_H")
+        self.h = X_H(initial, self.profiles[:,h_ind], self.scalars["s1_state"], 
+                   valid_initial, self.valid_profiles[:,h_ind], self.valid_scalars["s1_state"],
+                   IF_interpolator)
+
+        # instantiate and train density profile model
+        dens_ind = self.names.index("logRho")
+        self.dens = Density(initial,
+                       self.profiles[:,dens_ind],
+                       valid_initial,
+                       self.valid_profiles[:,dens_ind],
+                       IF_interpolator)
+        self.dens.train()        
+
+    def predict(self,inputs):
+        """Predict density and H mass fraction profiles from inputs.
+        Args:
+            inputs (array-like) : log-space initial conditions of N binaries to predict, shape (N,3).
+        Returns:
+            mass_coords (array-like) : linear-scale mass enclosed profile coordinates.
+            density_profiles (array_like) : log-scale density profile coordinates. 
+            h_profiles (array_like) : H mass fraction profile coordinates
+        """
+        mass_coords, density_profiles = self.dens.predict(inputs)
+        mass_coords, h_profiles = self.h.predict(inputs)
+        
+        return mass_coords, density_profiles, h_profiles
+        
+                    
+    def save(self, filename):
+        """Save complete profiles interpolation model.
+        Args:
+            filename (str) : path/name of '.pkl' file where the model will be saved.
+        """
+        SAVE_ATTRS = self.__dict__
+        DONT_SAVE = []
+        myattrs = {key: SAVE_ATTRS[key]
+                   for key in SAVE_ATTRS if key not in DONT_SAVE}
+
+        with open(filename, 'wb') as f:
+            pickle.dump(myattrs, f)
+
+    def load(self, filename):
+        """Load interpolation model, which can only be used for predictions.
+        Args:
+            filename (str) : path/name of '.pkl' file to be loaded.
+        """
+        with open(filename, 'rb') as f:
+            myattrs = pickle.load(f)
+            for key in myattrs:
+                setattr(self, key, myattrs[key])
+    
+    def mono_decrease(self,profiles):
+        """Enforce monotonicity in profiles such as density that must monotonically decrease
+        Args:
+            profiles (array-like) : N profiles to be post-processed for monotonicity
+        Returns:
+            profiles_mono (array-like) : post-processed profiles
+        """
+        
+        def mono_renorm(arr):
+            arr_copy = arr.copy()
+            # force rising points down
+            for i in range(1,len(arr_copy)):
+                if arr_copy[i]-arr_copy[i-1]>0:
```

Comment:
Minor: Why don't you write it as `if arr_copy[i]>arr_copy[i-1]:`? There shouldn't be a need to request an additional difference to be calculated.